### PR TITLE
isl: migrate to Conan v2

### LINKED ISSUE #14915: [package] ISL/0.24 lacks support for Conan 2.0
### Description

Uses old-style imports

### Package and Environment Details

isl/0.24

### Conan profile

Any

### Steps to reproduce

conan-2.0 build -pr:b tools.jinja -pr:h tools.jinja --name isl --version 0.24 . 

### Logs

<details><summary>Click to expand log</summary>

```
conan-2.0 build -pr:b tools.jinja -pr:h tools.jinja --name isl --version 0.24 .

-------- Input profiles --------
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
[options]
cmake*:bootstrap=True
[tool_requires]
!openssl*: cmake/3.24.2

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
[options]
cmake*:bootstrap=True
[tool_requires]
!openssl*: cmake/3.24.2

ERROR: Error loading conanfile at '/tmp/tmp.UuQlL1yi0W/isl/conanfile.py': Unable to load conanfile in /tmp/tmp.UuQlL1yi0W/isl/conanfile.py
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmp/tmp.UuQlL1yi0W/isl/conanfile.py", line 1, in <module>
    from conans import AutoToolsBuildEnvironment, ConanFile, tools
ImportError: cannot import name 'AutoToolsBuildEnvironment' from 'conans' (/mathworks/devel/sandbox/marc/conan-2.0/conan-io/conans/__init__.py)
```

</details>

### PR DESCRIPTION
**Pull request by valgur**
Minor additions to #14916. Notably, enabled `imath` support.

- Resolves #14916.
- Resolves #14915.

### CODE REVIEW

**Comment by uilianries** [comment]
@System-Arch hello! I see you are the co-author of this PR, could you please sign the CLA listed on the comment above: https://github.com/conan-io/conan-center-index/pull/21153#issuecomment-1813385987

File: recipes/isl/all/conanfile.py [hunk 1]
```diff
@@ -13,121 +16,109 @@ class IslConan(ConanFile):
...
-        if self.settings.compiler == "Visual Studio":
-            self.build_requires("automake/1.16.4")
+        if self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
+        if self.options.autogen:
+            self.tool_requires("autoconf/2.71")    # Needed for autoreconf
+            self.tool_requires("automake/1.16.5")  # Needed for aclocal called by autoreconf--does Coanan 2.0 need a transitive_run trait?
+            self.tool_requires("libtool/2.4.7")    # Needed for libtool
```
Comment (jwillikers) [inline recipes/isl/all/conanfile.py:75]: I'm not sure if all of these comments are necessary.

**Comment by valgur** [comment]
@uilianries Looks like @System-Arch has been inactive on GitHub since May. Since he/she has clearly signed an older version of the CLA for previous PRs, could a one-time exception be made, maybe?

**Comment by uilianries** [comment]
@valgur Unfortunately it's not possible. The CLA is not only mandatory, but part of the CI, and we don't have how to add exceptions.

What you could do is a push force and keep only you as the author.

**Review by jwillikers — APPROVED** [review]

**Review by AndreyMlashkin — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**